SunCalcNet/SunCalc.cs

```diff
@@ -91,10 +91,18 @@ public static IEnumerable<SunPhase> GetSunPhases(DateTime date, double lat, doub
             foreach (var sunPhase in SunPhaseAngle.List)
             {
                 var jset = SunTime.GetSetJ(sunPhase.Angle * Constants.Rad, lw, phi, dec, n, m, l);
-                var jrise = jnoon - (jset - jnoon);
 
-                sunPhaseCol.Add(new SunPhase(sunPhase.RiseName, jrise.FromJulian()));
-                sunPhaseCol.Add(new SunPhase(sunPhase.SetName, jset.FromJulian()));
+                if (!double.IsNaN(jset))
+                {
+                    var jrise = jnoon - (jset - jnoon);
+                    sunPhaseCol.Add(new SunPhase(sunPhase.RiseName, jrise.FromJulian()));
+                    sunPhaseCol.Add(new SunPhase(sunPhase.SetName, jset.FromJulian()));
+                }
+                else
+                {
+                    sunPhaseCol.Add(null);

```

Comment:
I wouldn't add any nulls to the sun phases collections. Sunrise/sunset never happen at that latitude and date and they don't need to be added to the collection. I would remove the whole else.

Reply:
I considered doing it that way because I don't like nulls either, but thought that would be a breaking change because I access phases in my code like this:
`sunPhases[SunPhaseName.NauticalDawn.Value].ToLocalTime();`
which no longer works for cases like the example [here](https://github.com/mourner/suncalc/issues/127), because if nulls are not used as filler values then dawn could be at index 2 instead of index 6.